.github/workflows/build-and-push-image.yml

```diff
@@ -4,8 +4,49 @@ on:
   workflow_dispatch:
   workflow_call:
 
+concurrency:
+  group: build-and-push-image-${{ github.sha }}
+
 jobs:
+  check-image-presence:
+    name: Check if images already exist
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
```

Comment:
Do we need this permission, I can't see it being used in this job?

Reply:
It's needed for the configure-aws-credentials action